src/components/Buttons/QRCodeButton.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// TODO: Extract to common shared UI library
+
+import { QrCode } from 'lucide-react'
+import QRCode from 'react-qr-code'
+import ReactTooltip from 'react-tooltip'
+import styled from 'styled-components'
+
+const QRCodeButton = ({ textToEncode }: { textToEncode: string }) => {
+  return (
+    <>
+      <StyledClipboardIcon data-tip data-for="qr-code-tooltip" data-event="click" size={15} />
+      <Tooltip id="qr-code-tooltip" backgroundColor="black" globalEventOff="click">
+        <QRCode size={150} value={textToEncode} bgColor="black" fgColor="white" />
+      </Tooltip>
+    </>
+  )
+}
+
+const StyledClipboardIcon = styled(QrCode)`
+  margin-left: 10px;
+  cursor: pointer;
+  color: ${({ theme }) => theme.font.secondary};
+`
+
+const Tooltip = styled(ReactTooltip)`
+  opacity: 1 !important;
+`
+
+export default QRCodeButton
```

Comment:
Same remark about style wrapping.

Reply:
Hm, I don't think your previous remark applies here, because we have 2 individual components, wrapped with a fragment. How would you change this code?

Reply:
If our goal is to systematically allow for subsequent styling (by wrapping the styled component), then I'd do:

```ts
const QRCodeButton = ({ textToEncode, className }: { textToEncode: string; className?: string }) => {
  return (
    <>
      <StyledClipboardIcon className={className} data-tip data-for="qr-code-tooltip" data-event="click" size={15} />
      <Tooltip id="qr-code-tooltip" backgroundColor="black" globalEventOff="click">
        <QRCode size={150} value={textToEncode} bgColor="black" fgColor="white" />
      </Tooltip>
    </>
  )
}

const StyledClipboardIcon = styled(QrCode)`
  margin-left: 10px;
  cursor: pointer;
  color: ${({ theme }) => theme.font.secondary};
`

const Tooltip = styled(ReactTooltip)`
  opacity: 1 !important;
`

export default styled(QRCodeButton)``
```

It encourages us to always pass the `className` down (linked to our previous discussion, last week I believe).

I'm still not 100% sure about this, it's a tradeoff between coherence (always being able to extend style) and added complexity.

Reply:
I see, you're right.

This is not needed: `export default styled(QRCodeButton)` and can be replaced by simply `export default QRCodeButton` (or even better, bringing it up to `export const QRCodeButton = ({...`)

Updating...